packages/gatsby-image/src/index.js

```diff
@@ -101,6 +107,53 @@ function getIO() {
   return io
 }
 
+function generateImageSources(imageVariants) {
+  return imageVariants.map(({ src, srcSet, srcSetWebp, media, sizes }) => (
+    <React.Fragment key={src}>
+      {srcSetWebp && (
+        <source
+          type="image/webp"
+          media={media}
+          srcSet={srcSetWebp}
+          sizes={sizes}
+        />
+      )}
+      <source media={media} srcSet={srcSet} sizes={sizes} />
+    </React.Fragment>
+  ))
+}
+
+function generateTracedSVGSources(imageVariants) {
+  return imageVariants.map(({ src, media, tracedSVG }) => (
+    <source key={src} media={media} srcSet={tracedSVG} />
```

Comment:
This issue is happening because of the spaces, in `srcset` you provide a list of urls as sources. If no descriptor(w or x unit) is provided, it defaults to 1x or 100w. List is separated by comma, and width descriptors when used by a space after the url/uri before the comma.

So base64 uri is working as there is no spaces, browser understands that fine. With the url encoded inlined xml, it hasn't encoded the spaces to %20, so it gets confused as it's not indicating a valid descriptor. This will work:

```
srcset="data:image/svg+xml,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20width%3D%27120%27%20height%3D%27180%27%3E%3Cpath%20d%3D%27M51%2042l-19%204-2-1c2-2-1-2-3%200-3%202-4%204-1%203l-1%203-3%203c-1%200-3%205-3%208l-1%204c-3%201-5%205-4%209%200%204%201%205%204%205%202%201%201%202-2%201-1%200-2%201-2%203l-4%206c-5%207-3%2016%206%2021l7%202c1-1%205%202%206%204%200%204%203%209%204%2010l3%207c1%206%204%2010%208%2011%201%201%204%202%204%204%203%203%206%201%204-3-1-2-1-5%201-5l3%206c0%203%201%205%202%205%209%202%2014%200%2021-6%206-5%2015-17%2018-22l3-4%203-4%203-4c2%200%205-6%206-13%201-8%200-10-5-17-2-3-3-5%200-5%201%201%201%201%201-1v-2c2-1-2-5-7-8-3-1-3-1-2%202v4l-1%203c1%201%200%202-1%202l-1-2c1-3-1-10-2-10l-1-1-2-5c-2-2-3-5-3-8%200-4-1-6-4-6l-4-1-10-3a514%20514%200%200%201-19%201%27%20fill%3D%27%23d3d3d3%27%20fill-rule%3D%27evenodd%27%2F%3E%3C%2Fsvg%3E"
```

But base64 would be less in length, so you probably want to use that when assigning the value to srcset(it's fine for regular src on img though).

No need for SVG guru :)

Reply:
Sweet @polarathene. You're now part of the svg guru team 😄 lets just encode spaces with %20 everytime. It's less breaking and should be mostly smaller in size.

https://codepen.io/tigt/post/optimizing-svgs-in-data-uris is a good read to see that svg is smaller even with encoding spaces.

Reply:
Oh, you started responding to my earlier comments. I just deleted two older ones that have been better addressed in the larger consolidation post I made, mostly for my benefit, the deleted comments no longer contributed anything, I write enough for others to read as is :sweat_smile:

Thanks for that link :) I had seen a small table comparison by the same author of the `mini-svg-data-uri` libraries npm page. Hadn't gotten around to updating my comment to no longer suggest base64 :sweat_smile: